### PR TITLE
Add ClickTooltip and HoverTooltip components to improve mobile UX

### LINKED ISSUE #5851: Add click-only Tooltip component for tooltips with links to improve UX
### Description
Implement a new click-only [Tooltip](https://github.com/Automattic/woocommerce-payments/blob/8884edeb0c2c533f9e7a3261e267d25e41cb9ef2/client/components/tooltip/index.tsx#L17) component specifically designed for tooltip content that includes links or other interactive elements. This new component will improve the user experience on both touch devices and desktops. The existing hover-based Tooltip component will remain available for use cases where it is preferred.

### Acceptance criteria
* Implement a new click-only [Tooltip](https://github.com/Automattic/woocommerce-payments/blob/8884edeb0c2c533f9e7a3261e267d25e41cb9ef2/client/components/tooltip/index.tsx#L17) component, extending the functionality of the existing [TooltipBase](https://github.com/Automattic/woocommerce-payments/blob/8884edeb0c2c533f9e7a3261e267d25e41cb9ef2/client/components/tooltip/tooltip-base.tsx#L167) component.
* The new component should handle click interactions exclusively.
* ~Replace all instances of the existing Tooltip component with the new click-only variant for tooltips containing links to documentation.~  
  Only new tooltips on the redesigned overview page will use the click-only variant for now: [see comment](https://github.com/Automattic/woocommerce-payments/pull/5862#issuecomment-1487571028).

### Components using Tooltip to audit for external links
- [x] https://github.com/Automattic/woocommerce-payments/blob/18ccaf89ca4962523063da1cde2a1f7fe402aea1/client/components/loadable-checkbox/index.js (hover tooltip)
- [x] https://github.com/Automattic/woocommerce-payments/blob/18ccaf89ca4962523063da1cde2a1f7fe402aea1/client/components/payment-methods-checkboxes/payment-method-checkbox.js (**click tooltip**)
- [x] https://github.com/Automattic/woocommerce-payments/blob/18ccaf89ca4962523063da1cde2a1f7fe402aea1/client/components/payment-methods-list/payment-method.js (**click tooltip**)
- [x] https://github.com/Automattic/woocommerce-payments/blob/18ccaf89ca4962523063da1cde2a1f7fe402aea1/client/settings/express-checkout/index.js (hover tooltip)
- [x] https://github.com/Automattic/woocommerce-payments/blob/18ccaf89ca4962523063da1cde2a1f7fe402aea1/client/settings/fraud-protection/advanced-settings/rule-toggle.js (hover tooltip)
- [x] https://github.com/Automattic/woocommerce-payments/blob/18ccaf89ca4962523063da1cde2a1f7fe402aea1/client/transactions/list/index.tsx (hover tooltip)

### Additional context
This new component is part of the ongoing Simplify Deposits UI project.

### PR DESCRIPTION
**Pull request by Jinksi**
Resolves #5851

> **Warning**
> Although this is being implemented as part of the Simplify Deposits UI project, this PR applies to all uses of the Tooltip component and changes are not hidden behind a feature flag.

#### Changes proposed in this Pull Request

This PR replaces the [Tooltip](https://github.com/Automattic/woocommerce-payments/blob/8884edeb0c2c533f9e7a3261e267d25e41cb9ef2/client/components/tooltip/index.tsx#L17) component with two new components:
  * ClickTooltip: for tooltip content that includes links or other interactive elements (documentation links). This tooltip will only display when clicked.
  * HoverTooltip: for any other tooltip content. This tooltip displays on hover events and/or click events.

**All existing instances of the `Tooltip` component have been replaced with `HoverTooltip`.**

Change #5805 will use the ClickTooltip component only for [new tooltips with links from the Overview page](https://github.com/Automattic/woocommerce-payments/pull/5862#issuecomment-1487571028).

### Testing instructions

1. Go to any page within WCPay where tooltips are used.
   - Go to **Payments → Settings → Enable new checkout experience** (first-time setup)    
      image
   - Go to **Payments → Settings → Payment Methods List** (after being enabled)  
      image
5. Ensure the tooltip content is displayed on hover and is hidden after a 600ms mouseout delay or when hovering another tooltip.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `5862` or branch name `add/5851-add-click-only-tooltip-component-for-tooltips-with-links-to-improve-ux` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=add/5851-add-click-only-tooltip-component-for-tooltips-with-links-to-improve-ux) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 3fd782995111af364935bee72063d0ede4ac1909
- Build time: 2023-03-30 07:13:24 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/components/tooltip/test/index.test.tsx [hunk 1]
```diff
@@ -82,54 +88,271 @@ describe( 'Tooltip', () => {
...
-			screen.queryByText( 'Tooltip 2 content' )
+			screen.queryByText( 'Tooltip content' )
 		).not.toBeInTheDocument();
-		expect( handleHide1Mock ).not.toHaveBeenCalled();
-		expect( handleHide2Mock ).not.toHaveBeenCalled();
+		expect( handleHideMock ).not.toHaveBeenCalled();
+	} );
+} );
+
+describe( 'Tooltips', () => {
```
Comment (Jinksi) [inline client/components/tooltip/test/index.test.tsx:239]: I've added a new test suite to ensure the 2 variants hide each other correctly when required.

File: client/components/tooltip/test/index.test.tsx [hunk 2]
```diff
@@ -82,54 +88,271 @@ describe( 'Tooltip', () => {
 		expect( handleHideMock ).toHaveBeenCalled();
 	} );
 
-	it( 'asks other Tooltips to hide, when multiple are opened', () => {
```
Comment (Jinksi) [inline client/components/tooltip/test/index.test.tsx:85]: This test has been replaced by the 'Tooltips' test suite below.

File: client/components/loadable-checkbox/index.js [hunk 3]
```diff
@@ -6,11 +6,15 @@
...
 import React, { useEffect, useState } from 'react';
 import { CheckboxControl, VisuallyHidden } from '@wordpress/components';
-import './style.scss';
 import classNames from 'classnames';
-import { useManualCapture } from 'wcpay/data';
 import { Icon, warning } from '@wordpress/icons';
-import Tooltip from '../tooltip';
+
+/**
+ * Internal dependencies
```
Comment (Jinksi) [inline client/components/loadable-checkbox/index.js:13]: In this file, I organised the "Internal Dependencies" to improve code quality.

File: client/components/tooltip/index.tsx [hunk 4]
```diff
@@ -62,4 +69,42 @@ const Tooltip: React.FC< TooltipProps > = ( {
 	);
 };
 
-export default Tooltip;
```
Comment (Jinksi) [inline client/components/tooltip/index.tsx:65]: Tooltip is no longer exposed as the default export.

HoverTooltip and ClickTooltip are now exported from this file.

The intention is to make it more explicit for future devs using these components.

**Review by james-allan — APPROVED** [review]
These changes look good to me and work as described. 

I've done an audit and there's only 1 remaining `<Tooltip>` component in use however it's the `@wordpress` component being used in that context.

I started doing a change by change comparison but I don't think I understand the decision to make some of the existing tooltips clickable. For instance the ones in `payment-method-checkbox.js` don't appear to have links or anything that would require it to be a clickable type. 🤔 

I haven't been able to find the flow to observe these Tooltips in the UI. 

I'm happy to approve though based on your judgement of whether they should be clickable or hover ones.

**Comment by haszari** [comment]
> I started doing a change by change comparison but I don't think I understand the decision to make some of the existing tooltips clickable. For instance the ones in payment-method-checkbox.js don't appear to have links or anything that would require it to be a clickable type. 🤔

That sounds like something to follow up - as I understand it we should only use the new clickable variant for the new tooltips (with links). If there are any that we're unsure about, @Jinksi please confirm with design (@elizaan36).

**Comment by Jinksi** [comment]
Thanks @james-allan and @haszari for your input! To prevent any accidental merging, I'm marking this PR as a draft.

> We should only use the new clickable variant for the new tooltips (with links) 

From [your comment on the original PR](https://github.com/Automattic/woocommerce-payments/pull/5805#discussion_r1145449991) @haszari, I gathered that we should be updating all existing WCPay tooltips with link content to the new click-only behaviour.

@elizaan36, should the new click-only variant be used solely for the tooltips with links on the new overview page? Or should we be using the new click-only variant for all WCPay tooltips with links?

**Comment by haszari** [comment]
> For instance the ones in payment-method-checkbox.js don't appear to have links or anything that would require it to be a clickable type. 🤔
> From https://github.com/Automattic/woocommerce-payments/pull/5805#discussion_r1145449991 @haszari, I gathered that we should be updating all existing WCPay tooltips with link content to the new click-only behaviour.

Yep that's my understanding – based on @james-allan comment above it sounds like _maybe_ there are some that it's not clear if they have a link or not. 

I guess the main thing is that we need to ensure we've tested/seen all the updated tooltips (PR author and PR reviewer).

**Comment by Jinksi** [comment]
> it sounds like _maybe_ there are some that it's not clear if they have a link or not. 

I will double-check the modified tooltips and ensure that only those with links are using the click-only variant. Thanks @james-allan for spotting!

**Comment by elizaan36** [comment]
> @elizaan36, should the new click-only variant be used solely for the tooltips with links on the new overview page? Or should we be using the new click-only variant for all WCPay tooltips with links?

hey, thanks for the ping on this. Let's use the click-only variant for just the new tooltips with links from the Overview page.

**Comment by Jinksi** [comment]
Thanks for confirming @elizaan36 🙌 

I've updated this PR to ensure all existing instances of the `Tooltip` component have been replaced with `HoverTooltip`.

`ClickTooltip` will only be used for tooltips with links on the Overview page in #5805.

Based on this @james-allan, are you happy to approve?

**Comment by james-allan** [comment]
> Based on this @james-allan, are you happy to approve?

Yep the changes look good to me in line with the advice above. 👍

**Review by james-allan — APPROVED** [review]

**Pull request merged**